File: Data Structures & Algorithms/valid-sudoku/submission-1.java
class Solution {
    public boolean isValidSudoku(char[][] board) {

        HashSet<String> set = new HashSet<>();

        for(int r = 0; r<board.length;r++){
            for(int c = 0; c<board[0].length;c++){
                int num = board[r][c];

                if(num=='.'){
                    continue;
                }
                if( !set.add(num+"row"+r) || !set.add(num+"col"+c) || !set.add(num + "box" + (r/3)+"-"+(c/3))){
                    return false;
                }
            }
        }

        return true;

      


        
    }
}

File: Data Structures & Algorithms/search-for-word/submission-1.py
class Solution:
    def exist(self, board: List[List[str]], word: str) -> bool:

        row = len(board)
        col = len(board[0])

        path = set()

        def back(r,c,i):
            if len(word)==i:
                return True
            
            if( r<0 or c < 0 or c>=col or r >=row or ((r,c) in path) or word[i] != board[r][c]):
                return False
            path.add((r,c))
            res = (back(r+1,c,i+1) or back(r-1,c,i+1) or back(r,c+1,i+1) or back(r,c-1,i+1))
            path.remove((r,c))
            return res

        

        for r in range(row):
            for c in range(col):

                if back(r,c,0):
                    return True
        return False

        
        